Automattic/wp-calypso
# Manage Purchase: Add notice to inform about available products for renewal

#### Changes proposed in this Pull Request

* Adds a new Notice to the Manage Purchase page that will be displayed when other upgrades from the same site are available for renewal.
![image](https://user-images.githubusercontent.com/15204776/81608657-3dd87a80-939c-11ea-8773-774bd0c0cd18.png)

* Inside the new notice, a new dialog can be opened that will display the detail of each available renewal, the user will be able to select which ones they one to proceed with.
![image](https://user-images.githubusercontent.com/15204776/81608774-68c2ce80-939c-11ea-9bda-49c072cde64e.png)

* Adds type definitions to format-currency

* Design discussion: pbOQVh-9b-p2

#### Testing instructions

* Go to a site that has multiple purchases expiring soon (within the next 3 months) or that are already expired and still renewable. (Turn auto-renew off to make sure they are marked as "expiring").
* Go to Manage Purchases and open any of the purchases that are expiring soon.
* You should see the new notice telling you that there are other products also expiring.
* Click "renew all" inside the notice to go to the checkout with all your products in the cart
* Go back and click on "other purchases", inside the notice click renew now to go to the checkout with all your products in the cart.
* Go back and click on "other purchases" inside the notice, unselect one of them and click renew now to go to the checkout with your selected products in the cart.
* Go back and click on "other purchases" inside the notice, click on one of the "manage purchase" links to go to that purchase detail page.
* Go back and click on "other purchases" inside the notice, unselect all the purchases and make sure that the "renew now" buttons becomes disabled, also make sure that the dialog can be closed using both the cancel button and by clicking in the backdrop.
